### PR TITLE
Fix reserved after reset

### PR DESCRIPTION
**Pull request by michalbiesek**
- if item was reserved add to freeq after reset
- rename _slab_evict_ok

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pelikan/72)

### CODE REVIEW

**Review by kasiawasiuta — COMMENTED** [review]
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/pmem/pelikan/72)* status: all files reviewed, 1 unresolved discussion (waiting on @jschmieg and @michalbiesek)

___
*[src/storage/slab/slab.c, line 206 at r1](https://reviewable.io/reviews/pmem/pelikan/72#-LmySWdE2eszYb_jHOBL:-LmySWdFDsegIh6_tHCF:b-zd07ky) ([raw file](https://github.com/pmem/pelikan/blob/dbc7c02dc938bdd031b021d4a895ae3a8f653fed/src/storage/slab/slab.c#L206)):*
> ```C
>         } else if (it->klen && !_slab_check_no_refcount(slab)) {
>            /* before reset item could be only reserved
>             * ensure that slab have a reserved item(s)
> ```

has reserved item(s)

**Review by michalbiesek — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/pelikan/72)* status: 1 of 2 files reviewed, 1 unresolved discussion (waiting on @jschmieg and @kasiawasiuta)

___
*[src/storage/slab/slab.c, line 206 at r1](https://reviewable.io/reviews/pmem/pelikan/72#-LmySWdE2eszYb_jHOBL:-LmyWiSKB8wbw9NuD0oA:b-896fix) ([raw file](https://github.com/pmem/pelikan/blob/dbc7c02dc938bdd031b021d4a895ae3a8f653fed/src/storage/slab/slab.c#L206)):*
<details><summary><i>Previously, kasiawasiuta wrote…</i></summary><blockquote>

has reserved item(s)
</blockquote></details>

Done.

File: src/storage/slab/slab.c [hunk 1]
```diff
@@ -162,6 +162,18 @@ _slab_put_item_into_freeq(struct item *it, uint8_t id)
     PERSLAB_INCR(id, item_free);
 }
 
+/*
+ * Because a slab can have a reserved item (claimed but not linked), which is
```
Comment (jschmieg) [inline src/storage/slab/slab.c:166]: Old comment should be updated to match new purpose of this function.

Comment (michalbiesek) [reply src/storage/slab/slab.c:166]: Done.

**Review by jschmieg — APPROVED** [review]

**Pull request merged**